Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/EntryDelegate.cpp

```diff
@@ -148,6 +149,80 @@ namespace AzToolsFramework
             return m_iconSize;
         }
 
+        void SearchEntryDelegate::paint(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const
+        {
+            auto data = index.data(AssetBrowserModel::Roles::EntryRole);
+            if (data.canConvert<const AssetBrowserEntry*>())
+            {
+                QStyleOptionViewItem opt(option);
+
+
+                bool isEnabled = (opt.state & QStyle::State_Enabled) != 0;
+                bool isSelected = (opt.state & QStyle::State_Selected) != 0;
+
+                opt.state &= ~QStyle::State_HasFocus;
+
+                // Get the hovered index in order to highlight the whole row.
+                AssetBrowserTableView* view = qobject_cast<AssetBrowserTableView*>(opt.styleObject);
+                QTableView::SelectionBehavior behavior = view->selectionBehavior();
+                QModelIndex hoveredIndex = view->GetHoveredIndex();
+                if (!isSelected && behavior != QTableView::SelectItems)
+                {
+                    if (behavior == QTableView::SelectRows && hoveredIndex.row() == index.row())
+                    {
+                        opt.state |= QStyle::State_MouseOver;
+                    }
+                }
+
+                QStyle* style = opt.widget ? opt.widget->style() : QApplication::style();
+
+                // draw the background
+                style->drawPrimitive(QStyle::PE_PanelItemViewItem, &opt, painter, opt.widget);
+
+                auto entry = qvariant_cast<const AssetBrowserEntry*>(data);
+                auto sourceEntry = azrtti_cast<const SourceAssetBrowserEntry*>(entry);
```

Comment:
suggestion: It might be clearer to move these variables closer to their first use (it's good to always try and keep this distance to a minimum)